lib/api/database.js

```diff
@@ -27,33 +27,73 @@ var Table = function(name) {
  *   MinRangeKey: the minimum range key to return, inclusive
  *   MaxRangeKey: the maximum range key to return, inclusive
  *   Limit: maximum items to return
+ *   ScanIndexForward: if false, will return results in  descending order. If true (default),
+ *     it will return results in ascending order
+ *   ConsistentRead: if false, the operation will use eventually consistent reads. If true (default),
+ *     the operation will use strongly consistent reads
  */
 Table.prototype.list = function(params, callback) {
   var expression = '#hashKey=:hashKey';
   var attributeNames = { '#hashKey': keyName(params.Key) };
   var attributeValues = { ':hashKey': keyValue(params.Key) };
+  var scanIndexForward = true;
+  var consistentRead = true;
   if (params.MinRangeKey && params.MaxRangeKey) {
     expression += ' AND #rangeKey BETWEEN :minRangeKey AND :maxRangeKey';
     attributeNames['#rangeKey'] = keyName(params.MinRangeKey);
     attributeValues[':minRangeKey'] = keyValue(params.MinRangeKey);
     attributeValues[':maxRangeKey'] = keyValue(params.MaxRangeKey);
-  } else if (params.minRangeKey) {
+  } else if (params.MinRangeKey) {
     expression += ' AND #rangeKey>=:minRangeKey';
     attributeNames['#rangeKey'] = keyName(params.MinRangeKey);
     attributeValues[':minRangeKey'] = keyValue(params.MinRangeKey);
-  } else if (params.maxRangeKey) {
+  } else if (params.MaxRangeKey) {
     expression += ' AND #rangeKey<=:maxRangeKey';
     attributeNames['#rangeKey'] = keyName(params.MaxRangeKey);
     attributeValues[':maxRangeKey'] = keyValue(params.MaxRangeKey);
   }
-
+  if (params.ScanIndexForward !== undefined) {
+    scanIndexForward = params.ScanIndexForward;
+  }
+  if (params.ConsistentRead !== undefined) {
+    consistentRead = params.ConsistentRead;
+  }
   this.dynamo.query({
     TableName: this.name,
     KeyConditionExpression: expression,
     ExpressionAttributeNames: attributeNames,
     ExpressionAttributeValues: attributeValues,
-    ConsistentRead: true,
-    ScanIndexForward: true,
+    ConsistentRead: consistentRead,
+    ScanIndexForward: scanIndexForward,
+    Limit: params.Limit || 100,
+  }, function(err, result) {
+    callback(err, err ? undefined : result.Items || []);
+  });
+};
+
+/**
+ * Lists items with a given global secondary index and optional range of range keys.
```

Comment:
It looks like you didn't implement the range key part (which is fine!). Can you remove the comment though?